networkx/algorithms/tournament.py

```diff
@@ -328,22 +328,25 @@ def two_neighborhood(G, v):
         out-neighbors of `v`), and the nodes at distance two.
 
         """
+        v_adj = G._adj[v]
         return {
-            x for x in G if x == v or x in G[v] or any(is_path(G, [v, z, x]) for z in G)
+            x
+            for (x, x_pred) in G._pred.items()
+            if x == v or x in v_adj or any(z in v_adj for z in x_pred)
         }
 
-    def is_closed(G, nodes):
+    def is_closed(G, S):
         """Decides whether the given set of nodes is closed.
 
         A set *S* of nodes is *closed* if for each node *u* in the graph
         not in *S* and for each node *v* in *S*, there is an edge from
         *u* to *v*.
 
         """
-        return all(v in G[u] for u in set(G) - nodes for v in nodes)
+        return all(u in S or v in G._adj[u] for u in G for v in S)
```

Comment:
I think this should be faster since it can short-circuit when `u in S`...
```suggestion
        return all(u in S or all(v in G._adj[u] for v in S) for u in G)
```

Reply:
Thanks for testing it :). Clearly this is not a performance-critical part of the code. I'm personally in favor of `return all(u in S or S <= G._adj[u].keys() for u in G)` because it's the one I find easiest to read, but all are good.